### PR TITLE
Add unit tests to the vegetation processes

### PR DESCRIPTION
**Pull request by mahabadri**
All the vegetation processes have been refactored so that the kernels are split into small functions, which are then tested by unit tests. Only the kernels in photosynthesis still need to be refactored and tested

### CODE REVIEW

**Comment by mahabadri** [comment]
Sure, I think photosynthesis might take some time, so maybe you can start reviewing the other processes and their corresponding tests?

**Comment by mahabadri** [comment]
Sorry, one moment, stomatal conductance is still missing, let me do that first

File: test/vegetation/carbon_dynamics_tests.jl [hunk 1]
```diff
@@ -1,44 +1,60 @@
...
-    @test λ_NPP = compute_λ_NPP(vegcarbon_dynamics, LAI_b) == 1.0
+    λ_NPP = compute_λ_NPP(vegcarbon_dynamics, LAI_b)
+    @test λ_NPP == 1.0
 end
 
 @testset "LAI_b test" begin
     vegcarbon_dynamics = PALADYNCarbonDynamics()
     # Test LAI_b should be positive for a positive C_veg
-    C_veg = 0.5
+    C_veg = 0.5 # Mock value
```
[the remark this answers is not in the document]
Comment (bgroenks96) [reply test/vegetation/carbon_dynamics_tests.jl:36]: I think I would prefer a thoughtful grid search including boundary conditions to randomly generated values.

[the remark this answers is not in the document]
Comment (mahabadri) [reply test/vegetation/carbon_dynamics_tests.jl:36]: Ok, I’ll change that across the tests. But, just to be sure I understand you correctly, @bgroenks96, by boundary condition you mean that for bounded variables (with a minimum and maximum value), I should do a grid search within that interval?

File: test/vegetation/vegetation_dynamics_tests.jl [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
     # Test ν == ν_seed (ν_star should be equal to ν_seed)
-    ν = vegetation_dynamics.ν_seed
-    ν_star = compute_ν_star(vegetation_dynamics, ν)
-    @test ν_star == vegetation_dynamics.ν_seed
-end
+    ν = veg_dynamics.ν_seed
+    ν_star = compute_ν_star(veg_dynamics, ν)
+    @test ν_star == veg_dynamics.ν_seed
+
+    # TODO ν should be between 0 and 1!
```
Comment (bgroenks96) [inline test/vegetation/vegetation_dynamics_tests.jl:29]: you mean it's broken? then add a `@test_broken`

**Review by bgroenks96 — CHANGES_REQUESTED** [review]
Looks great! Thanks, Maha! I think these simple tests are absolutely fine for now. It's important not to get too carried away with tests early on because they all need to be changed whenever the code gets refactored.

My only request is that for the `ν_star` test, if it's currently broken (i.e. the value is not what you expect), you should add the test and make it `@test_broken`. This makes it clear when we run the test that there is something that needs to be fixed, but still allows the tests to run and pass in the interim.

File: test/vegetation/carbon_dynamics_tests.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (mahabadri) [reply test/vegetation/carbon_dynamics_tests.jl:36]: Ok, I'll try to change that and then you can give me your feedback

File: test/vegetation/autotrophic_respiration_tests.jl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+@testset "NPP test" begin
+    autoresp = PALADYNAutotrophicRespiration()
+    # Test NPP should be the difference between GPP and Ra
+    GPP = 0.5 # Mock value
+    Ra = 0.3 # Mock value
+    NPP = compute_NPP(autoresp, GPP, Ra)
+    @test NPP == GPP - Ra
+
+    # TODO is NPP always positive ? GPP always > Ra?
```
[the remark this answers is not in the document]
Comment (mahabadri) [reply test/vegetation/autotrophic_respiration_tests.jl:67]: Yes, exactly, this is the case for many variables in the code. For now, there’s nothing in the code to prevent that from happening, so I’ve just left some comments for us to think about enforcing these physical bounds.

File: test/vegetation/vegetation_dynamics_tests.jl (hunk 2 above)
Comment (mahabadri) [reply test/vegetation/vegetation_dynamics_tests.jl:29]: Sorry, the comment is not clearly formulated here, the test does return the expected value. This comment is also meant for us to think about how to enforce these physical bounds.

**Comment by bgroenks96** [comment]
@mahabadri will submit further improvements to the tests in a subsequent PR. I think what's here is already good for now. I will merge the changes from main and then merge back.
